src/libstd/io/buffered.rs

```diff
@@ -417,17 +417,103 @@ impl<W> fmt::Display for IntoInnerError<W> {
     }
 }
 
-/// Wraps a Writer and buffers output to it, flushing whenever a newline
+/// Wraps a writer and buffers output to it, flushing whenever a newline
 /// (`0x0a`, `'\n'`) is detected.
 ///
-/// The buffer will be written out when the writer is dropped.
+/// The [`BufWriter`][bufwriter] struct wraps a writer and buffers its output.
+/// But it only does this batched write when it goes out of scope. Sometimes,
```

Comment:
This clause isn't quite true as it will write the internal buffer whenever it goes out of scope *or* it fills up the internal buffer (e.g. it's not infinitely buffered)